fix: "edit add labels" make labels handle -f flag same as commonLabels

fktym opened the pull request:
`kustomize edit add labels` command modifies labels to handle the -f flag the same way as commonLabels. This change should also address one of the concerns raised in [#4746 (comment)](https://github.com/kubernetes-sigs/kustomize/issues/4746#issuecomment-2237679544).

Current Behavior:
- commonLabels: `kustomize edit add labels name:value -f` overwrites existing labels when using -f flag
- labels: `kustomize edit add labels name:value -f --without-selector` creates a new Labels entry even with -f flag

Fixed Behavior:
- labels: `kustomize edit add labels name:value -f --without-selector` overwrites existing labels when using -f flag

Note:
- When determining if labels are the same, we also check if the includeSelector and includeTemplates values match

This change unifies the -f flag behavior between commonLabels and labels. I needed this overwrite functionality as I use kustomize edit add labels in CI to update versions.

Related:
    - https://github.com/kubernetes-sigs/kustomize/issues/4746#issuecomment-2237679544 
    - https://github.com/kubernetes-sigs/kustomize/pull/4486

stormqueen1990 commented:
/test all

stormqueen1990 reviewed (commented):
Hi there, @fktym! 👋🏻

Thank you for your contribution!

I just have a small suggestion on this for enhanced usability. Other than that, LGTM.

fktym commented on kustomize/commands/edit/add/addmetadata.go:
```diff
@@ -154,10 +149,67 @@ func (o *addMetadataOptions) addLabels(m *types.Kustomization) error {
+
+func (o *addMetadataOptions) writeToMapEntry(m map[string]string, k, v string, kind kindOfAdd) error {
+	if _, ok := m[k]; ok && !o.force {
+		return fmt.Errorf("%s %s already in kustomization file", kind, k)
```
@stormqueen1990 Thanks for the feedback! 
Totally agree that it'll help with usability. I've made the changes - mind taking another look?

varshaprasad96 reviewed (approved):
/lgtm
/approve

fktym commented:
@varshaprasad96 
Thanks for the LGTM! Sorry, it seems I changed the error message but forgot to update the test accordingly. [I've fixed it now](https://github.com/kubernetes-sigs/kustomize/compare/8aa849c961df2278294651337c1dfea463703e64..70e4d2e24c77b2be5e90ff62624ffc9eabe354c1), please review again

stormqueen1990 reviewed (commented):
/lgtm
/ok-to-test

Thanks @fktym!

fktym commented:
Thank you for the review. Looking forward to the release!
